### PR TITLE
Netbox node view plugin

### PR DESCRIPTION
**Pull request by drossy**
Initial commit of node plugin for Netbox support

### CODE REVIEW

File: node/node-netbox.inc.php [hunk 1]
```diff
@@ -0,0 +1,182 @@
...
+
+require_once('Horde/String.php');
+require_once('Horde/Text/Diff.php');
+require_once('Horde/Text/Diff/Engine/Native.php');
+require_once('Horde/Text/Diff/Op/Base.php');
+require_once('Horde/Text/Diff/Op/Add.php');
+require_once('Horde/Text/Diff/Op/Copy.php');
+require_once('Horde/Text/Diff/Op/Change.php');
+require_once('Horde/Text/Diff/Renderer.php');
+require_once('Horde/Text/Diff/Renderer/Inline.php');
```
Comment (jrha) [inline node/node-netbox.inc.php]: You can probably drop all of these as they were only used by the Overwatch data (which NetBox doesn't have).

File: node/node-netbox.inc.php [hunk 2]
```diff
@@ -0,0 +1,182 @@
...
+        } else {
+            echo "<p class=\"warning\">No rack power information.</p>\n";
+        }
+    }
+
+    function detail($NODE, $SHORT)
+    {
+        global $NETBOX_URL;
+
+        $renderer = new Horde_Text_Diff_Renderer_Inline();
```
Comment (jrha) [inline node/node-netbox.inc.php]: You can drop this too as you're not using it.

File: node/node-netbox.inc.php [hunk 3]
```diff
@@ -0,0 +1,182 @@
...
+							$found_supply=true;
+							$room_pdu=$this->get_netbox_info_by_id($conn['power_outlet']['device']['id']);
+		                	printf("<dt>%s (%s)</dt><dd>%s</dd>\n", $room_pdu['name'], $rack_pdu['name'], $room_pdu['site']['name']);
+						}
+					}
+					if ($found_supply == false) {
+						printf("<dt>Magic! (%s)</dt><dd>Unknown</dd>\n", $rack_pdu['name']);	
+					}
+				} else {
+					printf("<dt>Magic! (%s)</dt><dd>Unknown</dd>\n", $rack_pdu['name']);
```
Comment (jrha) [inline node/node-netbox.inc.php]: What does "Magic!" mean in these two contexts?
Is it worth distinguishing between them?

File: node/node-netbox.inc.php [hunk 4]
```diff
@@ -0,0 +1,182 @@
...
+	}
+	
+	// Get Netbox info for a single machine
+	// To the Netbox API this is technically a search
+	// So be careful we only get one response back	
+    private function get_netbox_info($machineName)
+    {
+        global $SQL;
+
+		$query=$this->netbox_search("/dcim/devices/",array("name"=>$SQL->real_escape_string($machineName)));
```
Comment (jrha) [inline node/node-netbox.inc.php]: Does this really need to be escaped for MySQL use with `real_escape_string`?
`$NODE` is already sanitised by the node wrapper.

Comment (drossy) [reply node/node-netbox.inc.php]: That was reused from the MagDB code

File: node/node-netbox.inc.php (hunk 3 above)
Comment (drossy) [reply node/node-netbox.inc.php]: I've clarified their meanings

File: node/node-netbox.inc.php (hunk 4 above)
Comment (jrha) [reply node/node-netbox.inc.php]: I know, but it's not being passed to a DB, so shouldn't need it.

**Comment by jrha** [comment]
Could you sort out the indentation? There seems to be a mix of tabs and spaces right now.

**Comment by drossy** [comment]
@jrha Can you review the latest commit?

**Review by jrha — APPROVED** [review]
LGTM

**Pull request merged**